### PR TITLE
Borrow responsive

### PR DESCRIPTION
**Pull request by Chris-CHEVALIER**

### CODE REVIEW

File: packages/frontend/components/Borrow/CollateralDropdown.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import { chains } from "../../machines/auth.machine"
+import Image from "next/image"
+
+type Chain = typeof chains[0]
+
+type CollateralDropdownProps = {
+  chains: Chain[]
+}
+
+export default function CollateralDropdown(props: CollateralDropdownProps) {
```
Comment (doliG) [inline packages/frontend/components/Borrow/CollateralDropdown.tsx:13]: I'm not sure about the cutting. I had the same concerns about the `settings` menu you created.

1. I'm not sure a component is neededbecause it contains no logic (presentational component) and will never be reused

2. A component is a reusable piece of UI. To me it make no sense to have only a dropdown without the button to trigger it 🤔 

3. It's not a Collateral Dropdown, but rather Chain Dropdown

File: packages/frontend/components/Borrow/LTVProgressBar.tsx [hunk 2]
```diff
@@ -32,21 +33,64 @@ export default function LTVProgressBar(props: LTVProgressBarProps) {
               sx={{
                 marginRight: "0.313rem",
                 fontSize: "0.875rem",
+                display: { xs: "none", sm: "inline" },
               }}
             />
           </Tooltip>
-          <Typography variant="xsmall">45% LTV (Recommended)</Typography>
+          <ClickableTooltip title="???" placement="top">
```
Comment (doliG) [inline packages/frontend/components/Borrow/LTVProgressBar.tsx:40]: Please add a todo here and ask Mark or Ivan what should we put as text :)

File: packages/frontend/components/Borrow/LTVProgressBar.tsx [hunk 3]
```diff
@@ -32,21 +33,64 @@ export default function LTVProgressBar(props: LTVProgressBarProps) {
...
+            <InfoOutlinedIcon
+              sx={{
+                marginRight: "0.313rem",
+                fontSize: "0.875rem",
+              }}
+            />
+          </ClickableTooltip>
+          <Typography
+            variant="xsmall"
+            sx={{ display: { xs: "none", sm: "inline" } }}
```
Comment (doliG) [inline packages/frontend/components/Borrow/LTVProgressBar.tsx:50]: Parent component already have `display: { xs: "none" }`, are you sure it's needed here ?

File: packages/frontend/components/Borrow/LTVProgressBar.tsx [hunk 4]
```diff
@@ -32,21 +33,64 @@ export default function LTVProgressBar(props: LTVProgressBarProps) {
...
+          >
+            45% LTV (Recommended)
+          </Typography>
+
+          <Typography
+            variant="xsmall"
+            sx={{ display: { xs: "inline", sm: "none" } }}
```
Comment (doliG) [inline packages/frontend/components/Borrow/LTVProgressBar.tsx:57]: Same remark as above

File: packages/frontend/components/Borrow/LTVProgressBar.tsx [hunk 5]
```diff
@@ -32,21 +33,64 @@ export default function LTVProgressBar(props: LTVProgressBarProps) {
...
         </Grid>
         <Grid item alignItems="center" marginRight="3rem">
-          <Typography variant="xsmall">75% LTV (MAX)</Typography>
           <Tooltip title="???" placement="top">
             <InfoOutlinedIcon
               sx={{
-                marginLeft: "0.313rem",
+                marginRight: "0.313rem",
                 fontSize: "0.875rem",
+                display: { xs: "none", sm: "inline" },
```
Comment (doliG) [inline packages/frontend/components/Borrow/LTVProgressBar.tsx:68]: Open question: do we need to specify `sm: inline` ? I think this is the default 🤔  Same remark for the 2 `Typography` below

File: packages/frontend/components/Borrow/LTVProgressBar.tsx [hunk 6]
```diff
@@ -32,21 +33,64 @@ export default function LTVProgressBar(props: LTVProgressBarProps) {
...
+          </Typography>
+
+          <Typography
+            variant="xsmall"
+            sx={{ display: { xs: "inline", sm: "none" } }}
+          >
+            SAFE LTV: 45%
+          </Typography>
         </Grid>
         <Grid item alignItems="center" marginRight="3rem">
```
Comment (doliG) [inline packages/frontend/components/Borrow/LTVProgressBar.tsx:62]: It make no sense to use `Grid item` without it's parent `Grid container` 🤔

File: packages/frontend/components/Borrow/Overview.tsx [hunk 7]
```diff
@@ -0,0 +1,231 @@
...
+import Image from "next/image"
+import PreviewTransaction from "./PreviewTransaction"
+import ClickableTooltip from "../Layout/ClickableTooltip"
+
+export default function Overview() {
+  const { palette } = useTheme()
+  const [showProvider, setShowProvider] = useState(false)
+
+  return (
+    <Container
```
Comment (doliG) [inline packages/frontend/components/Borrow/Overview.tsx:29]: Container should only be used as a root of the layout, see documentation https://mui.com/material-ui/react-container/#main-content

Here using a `Box` is a better fit

File: packages/frontend/components/Borrow/PreviewTransaction.tsx [hunk 8]
```diff
@@ -0,0 +1,311 @@
...
+  const isOpen = Boolean(anchorEl)
+
+  const openMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
+    setAnchorEl(event.currentTarget)
+  }
+
+  const closeMenu = () => {
+    setAnchorEl(null)
+  }
+  console.log(isOpen)
```
Comment (doliG) [inline packages/frontend/components/Borrow/PreviewTransaction.tsx:40]: remove console log

File: packages/frontend/components/Borrow/PreviewTransaction.tsx [hunk 9]
```diff
@@ -0,0 +1,311 @@
...
+
+  return (
+    <Container
```
Comment (doliG) [inline packages/frontend/components/Borrow/PreviewTransaction.tsx:43]: same remark as above I think you can simply remove it and put padding on `Grid` above

File: packages/frontend/components/Borrow/PreviewTransaction.tsx [hunk 10]
```diff
@@ -0,0 +1,311 @@
...
+import LTVProgressBar from "./LTVProgressBar"
+import ClickableTooltip from "../Layout/ClickableTooltip"
+
+export default function PreviewTransaction() {
+  const { palette } = useTheme()
+  const [showProvider, setShowProvider] = useState(false)
+  const [anchorEl, setAnchorEl] = React.useState<null | HTMLElement>(null)
+  const isOpen = Boolean(anchorEl)
+
+  const openMenu = (event: React.MouseEvent<HTMLButtonElement>) => {
```
Comment (doliG) [inline packages/frontend/components/Borrow/PreviewTransaction.tsx:33]: I think the naming can be improved as long as we are not opening a `menu` but rather previewing transaction 🤔

File: packages/frontend/components/Borrow/PreviewTransaction.tsx [hunk 11]
```diff
@@ -0,0 +1,311 @@
...
+                    <Grid container justifyContent="space-between">
+                      <Typography variant="smallDark">
+                        Collateral Provided
+                      </Typography>
+                      <Typography variant="small">1ETH (~1800.00)</Typography>
+                    </Grid>
+                  </Grid>
+
+                  <Grid item>
+                    <Grid container justifyContent="space-between">
```
Comment (doliG) [inline packages/frontend/components/Borrow/PreviewTransaction.tsx:128]: Can't you put justifyContent on the parent `Grid item` ? Not sure just asking

File: packages/frontend/components/Borrow/PreviewTransaction.tsx [hunk 12]
```diff
@@ -0,0 +1,311 @@
...
+import InfoOutlinedIcon from "@mui/icons-material/InfoOutlined"
+import AddCircleIcon from "@mui/icons-material/AddCircle"
+import CancelIcon from "@mui/icons-material/Cancel"
+import ExpandMoreIcon from "@mui/icons-material/ExpandMore"
+import Image from "next/image"
+
+import LTVProgressBar from "./LTVProgressBar"
+import ClickableTooltip from "../Layout/ClickableTooltip"
+
+export default function PreviewTransaction() {
```
Comment (doliG) [inline packages/frontend/components/Borrow/PreviewTransaction.tsx:27]: General thought with this component: I think we overcomplicated it with a lot of nested `Grid` (your bff haha).

This is something we should discuss directly cause it's difficult to explain by message 🤔

File: packages/frontend/components/Layout/ClickableTooltip.tsx [hunk 13]
```diff
@@ -0,0 +1,45 @@
...
+    | "left-end"
+    | undefined
+}
+
+export default function ClickableTooltip(props: ClickableTooltipProps) {
+  const [open, setOpen] = React.useState(false)
+
+  const handleTooltipClose = () => {
+    setOpen(false)
+  }
```
Comment (doliG) [inline packages/frontend/components/Layout/ClickableTooltip.tsx:28]:
```suggestion
  const handleTooltipClose = () => setOpen(false)
```

File: packages/frontend/components/Layout/ClickableTooltip.tsx [hunk 14]
```diff
@@ -0,0 +1,45 @@
...
+
+  const handleTooltipOpen = () => {
+    setOpen(true)
+  }
```
Comment (doliG) [inline packages/frontend/components/Layout/ClickableTooltip.tsx:32]:
```suggestion
  const handleTooltipOpen = () => setOpen(true)
```

File: packages/frontend/pages/_app.tsx [hunk 15]
```diff
@@ -33,6 +33,7 @@ function MyApp({ Component, pageProps }: AppProps) {
       }}
     >
       <ThemeProvider theme={theme}>
+        <div className="backdrop"></div>
```
Comment (doliG) [inline packages/frontend/pages/_app.tsx:36]: What is this ?

File: packages/frontend/styles/components/Borrow.module.css [hunk 16]
```diff
@@ -32,3 +32,9 @@
   opacity: 0.5;
   margin-right: 0.5rem;
 }
+
+.btn {
```
Comment (doliG) [inline packages/frontend/styles/components/Borrow.module.css:36]: Don't you think this should be a `variant` of our button in the theme ? 🤔

**Review by doliG — CHANGES_REQUESTED** [review]
Hello,

There are several problem because of fixed size (i.e `px` instead of `%` or fluid `Grid`) When I am in pure mobile view it's ok, but when I swith to tablet it become... meh:

image

Otherwise a lot of small stuff but nothing big.

Also please ensure precommit and formatting are working fine next time you request a review. I know the setup can be a little painful but then it shoud work so well that we'll forget about it :)

File: packages/frontend/components/Borrow/CollateralDropdown.tsx (hunk 1 above)
Comment (Chris-CHEVALIER) [reply packages/frontend/components/Borrow/CollateralDropdown.tsx:13]: Resolved by talking directly

File: packages/frontend/components/Borrow/PreviewTransaction.tsx (hunk 12 above)
Comment (Chris-CHEVALIER) [reply packages/frontend/components/Borrow/PreviewTransaction.tsx:27]: Unless I'm mistaken, `<Grid item>` doesn't accept `justify-content` prop.
To fix it, I replaced it by a `sx={{ display: "flex", justify-content: "center" }}`. Not sure to be a better way...

File: packages/frontend/styles/components/Borrow.module.css (hunk 16 above)
Comment (Chris-CHEVALIER) [reply packages/frontend/styles/components/Borrow.module.css:36]: Just add these styles in `primary` and `gradient` `variants`.
But you removed it in `chris/feat/screen-integration` merge request (https://github.com/Fujicracy/fuji-v2/commit/67c3dcee2bec9f971a2949bfc9d4d413ad5111fc) last week. We can talk more about it directly later.

**Review by doliG — APPROVED** [review]

**Pull request merged**